processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -63,13 +63,6 @@
   public AggregatorFactory getCombiningFactory();
 
   /**
-   * Gets a list of all columns that this AggregatorFactory will scan
-   *
-   * @return AggregatorFactories for the columns to scan of the parent AggregatorFactory
-   */
-  public List<AggregatorFactory> getRequiredColumns();
```

Comment:
removing this method is not backwards compatible for anyone providing their own aggregations as an extensions, and may cause things to fail at runtime if getRequiredColumns is called anywhere else in that extension.